src/execution_algorithms/ltfb/truncation_selection_exchange.cpp

```diff
@@ -102,18 +102,18 @@ EvalType TruncationSelectionExchange::evaluate_model(model& m,
   dc.collect_background_data_fetch(original_mode);
 
   // Can use validation if it is global
-  if (!dc.is_execution_mode_valid(execution_mode::testing)) {
+  if (!dc.is_execution_mode_valid(execution_mode::tournament)) {
     LBANN_ERROR("LTFB truncation selection requires ",
-                to_string(execution_mode::testing),
+                to_string(execution_mode::tournament),
                 " execution mode");
   }
 
   // Mark the data store as loading - Note that this is a temporary fix
   // for the current use of the tournament
-  m.mark_data_store_explicitly_loading(execution_mode::testing);
+  m.mark_data_store_explicitly_loading(execution_mode::tournament);
 
-  // Evaluate model on test (or validation?) set
-  get_trainer().evaluate(&m, execution_mode::testing);
+  // Evaluate model on validation set
+  get_trainer().evaluate(&m, execution_mode::tournament);
```

Comment:
One potential problem I see here is that tournament set are typically partitioned between trainers, TSE expect a global dataset for tournament (validation) to rank all trainers.

Reply:
The user should be responsible for configuring the dataset correctly. Within LBANN, there is no functional difference between the validation, test, and tournament sets. They're just labels to indicate usage. So we should use the validation set for validation, the test set for testing, and the tournament set for tournaments. It's confusing if tournament set isn't used for tournaments.